src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -70,6 +71,13 @@ def generate_payload(self):
 
         events.append({'name': self.event_name, 'properties': user_task})
 
+        for extension_event in self.extension_events:
+            if 'name' in extension_event and 'properties' in extension_event:
```

Comment:
Can this be verified in the `add_extension_events` instead of here? The `add_extension_events` signature can be changed to: `def add_extension_events(name, properties)` since the name and properties are always required.

Reply:
Fixed!